modules/caddyhttp/reverseproxy/reverseproxy.go

```diff
@@ -383,24 +399,51 @@ func (h *Handler) Provision(ctx caddy.Context) error {
 // Cleanup cleans up the resources made by h.
 func (h *Handler) Cleanup() error {
 	// close hijacked connections (both to client and backend)
-	var err error
-	h.connectionsMu.Lock()
-	for _, oc := range h.connections {
-		if oc.gracefulClose != nil {
-			// this is potentially blocking while we have the lock on the connections
-			// map, but that should be OK since the server has in theory shut down
-			// and we are no longer using the connections map
-			gracefulErr := oc.gracefulClose()
-			if gracefulErr != nil && err == nil {
-				err = gracefulErr
+	closeConnections := func() error {
+		var err error
+		h.connectionsMu.Lock()
+		defer h.connectionsMu.Unlock()
+
+		for _, oc := range h.connections {
+			if oc.gracefulClose != nil {
+				// this is potentially blocking while we have the lock on the connections
+				// map, but that should be OK since the server has in theory shut down
+				// and we are no longer using the connections map
+				gracefulErr := oc.gracefulClose()
+				if gracefulErr != nil && err == nil {
+					err = gracefulErr
+				}
+			}
+			closeErr := oc.conn.Close()
+			if closeErr != nil && err == nil {
+				err = closeErr
 			}
 		}
-		closeErr := oc.conn.Close()
-		if closeErr != nil && err == nil {
-			err = closeErr
+		return err
+	}
```

Comment:
This could probably be a separate function (unexported, but doesn't need to be nested inside Cleanup as far as I can tell.)

Reply:
Done. While looking for the best place where to put it I also split the `Cleanup` method and moved the parts that are related to hijacked connection into `streaming.go` so that the hijacked connection handling methods are close together.